sdk/Sdk.Generators/FunctionMetadataProviderGenerator.cs

```diff
@@ -0,0 +1,841 @@
+        {
+            if (attribute.AttributeClass != null && // this class is the exact type of binding/trigger (QueueTrigger, Queue input binding, etc)
+                attribute.AttributeClass.BaseType != null && // this base type tells you if something is input or output binding
+                attribute.AttributeClass.BaseType.BaseType != null) // this base type is the binding attribute type
+            {
+                return String.Equals(attribute.AttributeClass.BaseType.BaseType.GetFullName(), Constants.BindingAttributeType);
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Determines whether an attribute is a Function OutputBinding or not. This method assumes that any Function Binding related attributes passed in are
+        /// of the highest level (specific bindings like QueueTrigger, Queue Input Binding, etc).
+        /// </summary>
+        /// <param name="attribute">The exact attribute decorating a method/parameter/property declaration.</param>
+        private static bool IsOutputBindingAttribute(AttributeData attribute)
+        {
+            if (attribute.AttributeClass != null &&
+                attribute.AttributeClass.BaseType != null)
+            {
+                return String.Equals(attribute.AttributeClass.BaseType.GetFullName(), Constants.OutputBindingAttributeType);
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Determines whether an attribute is an HttpTrigger or not. This method assumes that any Function Binding related attributes passed in are
+        /// of the highest level (specific bindings like QueueTrigger, Queue Input Binding, etc).
+        /// </summary>
+        /// <param name="attribute">The exact attribute decorating a method/parameter/property declaration.</param>
+        private static bool IsHttpTrigger(AttributeData attribute)
+        {
+            if (attribute.AttributeClass != null)
+            {
+                return String.Equals(attribute.AttributeClass.GetFullName(), Constants.HttpTriggerBindingType);
+            }
+
+            return false;
+        }
+
+        private static bool IsEventHubsTrigger(AttributeData attribute)
+        {
+            if (attribute.AttributeClass != null)
+            {
+                return String.Equals(attribute.AttributeClass.GetFullName(), Constants.EventHubsTriggerType);
+            }
+
+            return false;
+        }
+        private static string? GetDataTypeFromType(string typeFullName)
+        {
+            string? dataType = null;
+
+            if (IsStringType(typeFullName))
+            {
+                dataType = "\"String\"";
+            }
+            // Is binary parameter type
+            else if (IsBinaryType(typeFullName))
+            {
+                dataType = "\"Binary\"";
+            }
+
+            return dataType;
+        }
+
+        private static bool IsStringType(string fullName)
```

Comment:
Consider moving these to `ISymbolExtensions` and work off of `ITypeSymbol` instead.